stm32-modules/heater-shaker/firmware/host_comms_task/freertos_comms_task.cpp

```diff
@@ -132,6 +148,15 @@ void run(void *param) {  // NOLINT(misc-unused-parameters)
                     local_task->tx_buf.committed()->data()),
                 tx_end - local_task->tx_buf.committed()->data());
             USBD_CDC_TransmitPacket(&_local_task.usb_handle);
+            if (UartReady) {
+                UartReady = false;
```

Comment:
More of a question: does sending data on the UART interfere with debugging operation at all? I.e. if you're debugging the Heater Shaker with an STLink instead of using it as a UART bridge, does the debugger work fine?

Reply:
We can actually debug while sending and receiving data over UART simultaneously!